src/database/declarations.h

```diff
@@ -1,17 +1,56 @@
-#ifndef TABLECOLUMNLITERALS_H
-#define TABLECOLUMNLITERALS_H
+#ifndef DECLARATIONS_H
+#define DECLARATIONS_H
 
-#include <QString>
-#include <QStringLiteral>
+#include <QtCore>
 
-// [F]: These constants deviate slightly from const naming convention to reflect their sql column name.
+/*!
+ * \brief An alias for QString
+ *
+ * Very long description *with* **markdown?**
+ * - - -
+ * # Header
+ */
+using ColName = QString;
+using ColData = QVariant;
```

Comment:
I dont see any reason to complicate this. Everything from the database is a string anyway. Because you have it as a variant you make the code in the entries needlessly complex by "casting" it to `toString` all the time.

Reply:
On the contrary, having this element as a QVariant actually avoids quite a lot of type casting. Most of the fields in the database are actually integers, not strings. The database holds different types, integers, text, numerics etc. If we cast all those into QString, we have to convert them back to their actual type before using them, effectively casting for example int to qstring to int.

There are also benefits when entering data into the database. When using QVariant, we can directly enter integers or strings depending on the database type into the map. Then, when creating the query to submit it, QVariant automatically binds the correct type to the insert query.

For example:

```cpp
// NewTailDialog
new_data.insert(DB_TAILS_MULTIPILOT, ui->operationComboBox->currentIndex() - 1); 
// here we can directly insert the integer instead of doing:
new_data.insert(QString::number(DB_TAILS_MULTIPILOT, ui->operationComboBox->currentIndex() - 1)); 
// here we would cast an integer to string, only to bind it to a string when we are creating a query, send a string to the database, and convert it to an integer that's stored in the database again.

// more examples
// NewFlightDialog
newData.insert(DB_FLIGHTS_TBLK, block_minutes);
newData.insert(DB_FLIGHTS_ACFT, tailsIdMap.value(ui->acftLineEdit->text()));

// ADatabase.cpp
bool ADatabase::insert(AEntry new_entry) {
query.prepare("statement with ? placeholders");
query.addbindvalue; 
// binds int to int and string to string instead of sending all strings to the database, which then has to do the casting
query.exec();
}

ADatabase::getEntryData(DataPosition data_position) {
// select data from database
//...
RowData entry_data;

    for (const auto &column : tableColumns.value(data_position.first)) {
        entry_data.insert(column, select_query.value(column));
    }
// QSqlQuery returns a QVariant already, so creating the map like this is much faster than doing this
RowData entry_data;

    for (const auto &column : tableColumns.value(data_position.first)) {
        entry_data.insert(column, select_query.value(column).toString);
    }
}
```

The same is true the other way around, when retreiving data from the database. Retreiving integers only to cast them to strings, and then do `string.toInt()` again seems rather wasteful and unneccessary. A `QMap<QString, QVariant>` is thus more efficient and also a more accurate representation of a database entry in my opinion.